specs/StatusBarMessage.md

```diff
@@ -0,0 +1,106 @@
+<!-- USAGE
+  * Fill in each of the sections (like Background) below
+  * Wrap code with `single line of code` or ```code block```
+  * Before submitting, delete all <!-- TEMPLATE marked comments in this file,
+    and the following quote banner:
+-->
+# Background
+The browser has a Status bar that displays text when hovering over a link, or performing some activity. Currently, 
+developers are able to opt in to disable showing the status bar through the browser 
+settings.
+
+Developers would also like to be able to opt in to intercept the messages which would 
+normally be displayed by the Status bar, and show it using thier own custom UI. 
+# Description
+We propose a new event for WebView2 that would allow developers to 
+listen for Status bar updates which are triggered by activity on the WebView, and then handle those updates however they want in their applications.
+
+Developers will be able to register an event handler for changes to the status bar message.
+# Examples
+## Win32 C++ Registering a listener for status bar message changes
+```
+CHECK_FAILURE(m_webView->add_StatusBarMessageChanged(
+    Microsoft::WRL::Callback<ICoreWebView2StatusBarMessageChangedEventHandler>(
+    [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT
+    {
+        
+        LPWSTR value;
+        CHECK_FAILURE(sender->get_StatusBarMessage(&value));
+        if (wcslen(value) != 0)
+        {
+
+            m_statusBar.show(value);
+        } else {
+            
+            m_statusBar.hide();
+        }
+
+        return S_OK;
+    }
+).Get(),
+&m_statusBarMessageChangedToken));
+```
+## .NET / WinRT Registering a listener for status bar message changes
+```
+webView.CoreWebView2.StatusBarMessageChanged += (CoreWebView2 sender, Object arg) =>
```

Comment:
The args should have a type

Reply:
There's no event args in this case because its a property changed event where the value of the new property is on the sender. Should we add an empty event args class for this? Previously I believe we've not been doing that.

Reply:
The guideline is to have a new type with no members, then pass null for the value at runtime. This provides the ability to version the args over time.

Reply:
OK, we'll need to add an empty ICoreWebView2StatusBarTextChanged (or whatever is correct after name changes to property) interface.